### PR TITLE
feat: inititalise chunk/<offset> endpoint

### PR DESCRIPTION
**Pull request by hlolli**
feat: initialise chunk/<offset> endpoint and unify chunk POST route

- Add a new GET route at `/chunk/:offset` (constrained to positive integers)
  to retrieve chunks by offset
- Create `createChunkOffsetHandler` to validate offset and return chunk data
- Implement SQL query to find matching transaction by offset (`offset.sql`)
  and extend `StandaloneSqliteDatabase` with `getTxByOffset`
- Export `chunkDataSource` from `system.ts` for use in the router
- Add TypeScript interface `TxByOffsetResult` for structured response typing
  within `standalone-sqlite.ts`
- Provide full test coverage in `handlers.test.ts` using `supertest` and `mock`
  (including error cases)
- Remove the `arweaveRouter` in favor of a unified chunk router (`chunkRouter`)
- Introduce `createChunkPostHandler` (migrating chunk POST logic from the old
  `arweaveRouter`)
- Register the POST route at `/chunk` (using the same router) and ensure chunk
  size limits via `json` parser
- Eliminate `offset.sql` (now handled inline within `standalone-sqlite.ts`)
- Remove duplicate `TxByOffsetResult` definition from `types.d.ts`,
  consolidating in `standalone-sqlite.ts`
- Refactor `app.ts` to drop the unused `arweaveRouter` and unify chunk routes
  under the `chunkRouter`

This commit initializes the chunk retrieval capability by offset and standardizes
all chunk-related endpoints (both GET and POST) within the same router,
simplifying configuration and improving maintainability.

### CODE REVIEW

File: src/database/standalone-sqlite.ts [hunk 1]
```diff
@@ -880,6 +881,26 @@ export class StandaloneSqliteDatabaseWorker {
     });
   }
 
+  getTxByOffset(offset: number): TxByOffsetResult {
```
Comment (djwhitt) [inline src/database/standalone-sqlite.ts:884]: The offset comparison logic is tricky enough to warrant unit tests. In particular, we should test the boundary conditions, offsets at the beginning and end of the transaction.

File: src/routes/chunk/handlers.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    let chunkBase64Url: string | undefined = undefined;
+    try {
+      chunkBase64Url = toB64Url(chunkData.chunk);
+    } catch (error) {
+      response.status(500).send('Error converting chunk to base64url');
+      return;
+    }
+
+    // this is a very limited interface of the node network chunk response
+    response.status(200).json({ chunk: chunkBase64Url });
```
Comment (djwhitt) [inline src/routes/chunk/handlers.ts:93]: We can hold off adding `tx_path` for now, but we should include the rest of the chunk metadata.

File: src/database/sql/core/offset.sql [hunk 3]
```diff
@@ -0,0 +1,9 @@
+-- selectStableTransactionOffsetById
```
Comment (djwhitt) [inline src/database/sql/core/offset.sql:1]: Minor style nit: I would call the file `offsets.sql` for consistency.

**Review by djwhitt — COMMENTED** [review]
General comment - the commit message should include a bit more information about the changes.

File: src/routes/chunk/constants.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// To add a GET route for /chunk/:offset where :offset is restricted to a positive integer,
+// we can use a regular expression in your route path to constrain :offset.
+export const CHUNK_OFFSET_PATH = '/chunk/:offset(\\d+)';
```
[the remark this answers is not in the document]
Comment (hlolli) [reply src/routes/chunk/constants.ts:21]: seems to be ok, it was more of an issue from index.ts not handler.ts, this fix will be included in next force push

File: src/database/sql/core/offset.sql (hunk 3 above)
Comment (hlolli) [reply src/database/sql/core/offset.sql]: done

File: src/routes/chunk/handlers.ts (hunk 2 above)
Comment (hlolli) [reply src/routes/chunk/handlers.ts]: ok done, I added from chunk metadata data_path

File: src/database/standalone-sqlite.ts (hunk 1 above)
Comment (hlolli) [reply src/database/standalone-sqlite.ts:890]: I did test this on png images and they came out correct. I can see how this can be problematic if the chunk offset requested in the http request for falls on some boundary. From unit test point of view, to give that a fair test, I'll want to run sql query in the unit test, with various offsets.

File: src/database/sql/core/offset.sql [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+SELECT id, data_root, offset, data_size
+FROM stable_transactions
+WHERE offset >= @offset
+  AND format = 2
+  AND data_size > 0
+  AND (offset - data_size) <= @offset
```
[the remark this answers is not in the document]
Comment (hlolli) [reply src/database/sql/core/offset.sql]: fixed

File: src/routes/chunk/constants.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (hlolli) [reply src/routes/chunk/constants.ts]: fixed

File: src/routes/chunk/handlers.test.ts [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    );
+
+    await request(app)
+      .get('/chunk/1234')
+      .expect(404)
+      .then((res: any) => {
+        assert.strictEqual(res.status, 404);
+        assert.strictEqual(res.text, 'Not Found');
+      });
+  });
```
[the remark this answers is not in the document]
Comment (hlolli) [reply src/routes/chunk/handlers.test.ts:126]: good idea

File: src/routes/chunk/handlers.test.ts [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+
+    await request(app)
+      .get('/chunk/1234')
+      .expect(404)
+      .then((res: any) => {
+        assert.strictEqual(res.status, 404);
+      });
+  });
+
+  it.only('should return 500 if chunk data source returns undefined', async () => {
```
[the remark this answers is not in the document]
Comment (hlolli) [reply src/routes/chunk/handlers.test.ts]: good catch, fixed

File: src/database/standalone-sqlite.ts (hunk 1 above)
Comment (hlolli) [reply src/database/standalone-sqlite.ts:890]: unit tests testing the boundry have been added https://github.com/ar-io/ar-io-node/pull/370/commits/bef4bce273c066275ef73463a2aa1e20dd638a67

**Review by djwhitt — COMMENTED** [review]
Found one likely issue based on the unit tests and added one minor suggestion about moving the route constant (again, sorry...).

File: src/routes/chunk/handlers.ts [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+  ChunkDataByAnySource,
+  ChunkMetadata,
+  ChunkMetadataByAnySource,
+} from '../../types.js';
+import { ArweaveCompositeClient } from '../../arweave/composite-client.js';
+import { Logger } from 'winston';
+
+// To add a GET route for /chunk/:offset where :offset is restricted to a positive integer,
+// we can use a regular expression in your route path to constrain :offset.
+export const CHUNK_OFFSET_PATH = '/chunk/:offset(\\d+)';
```
[the remark this answers is not in the document]
Comment (hlolli) [reply src/routes/chunk/handlers.ts:42]: ok, I'll duplicate it in the test file, otherwise it will import system.ts from index.ts. I think it's ok in this case.

File: src/routes/chunk/constants.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+/**
```
[the remark this answers is not in the document]
Comment (hlolli) [reply src/routes/chunk/constants.ts]: removed

File: src/database/standalone-sqlite.test.ts [hunk 10]
```diff
@@ -255,6 +255,104 @@ describe('StandaloneSqliteDatabase', () => {
...
+      assert.equal(txByOffsetResult1.id, undefined);
+      const txByOffsetResult2 = await db.getTxByOffset(49);
+      assert.equal(txByOffsetResult2.id, undefined);
+      // if at 50 until end of data_size (which is 50) to <= 100, should return tx1id
+      const txByOffsetResult3 = await db.getTxByOffset(50);
+      assert.equal(txByOffsetResult3.id, tx1id);
+      const txByOffsetResult4 = await db.getTxByOffset(99);
+      assert.equal(txByOffsetResult4.id, tx1id);
+      const txByOffsetResult5 = await db.getTxByOffset(100);
+      assert.equal(txByOffsetResult5.id, tx1id);
```
[the remark this answers is not in the document]
Comment (hlolli) [reply src/database/standalone-sqlite.test.ts:339]: fixed

File: src/routes/chunk/handlers.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (hlolli) [reply src/routes/chunk/handlers.ts]: implemented

**Review by djwhitt — APPROVED** [review]
